### PR TITLE
Finished the project

### PR DESCRIPTION
**Pull request by kjanos89**
Followed Pablo's tutorial video and finished the project.

### CODE REVIEW

File: Drinks.kjanos89/Drinks.kjanos89/Drinks.kjanos89/ApiController.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                Menu.ShowData(categories, "Categories");
+
+            }
+            else
+            {
+                Console.WriteLine("Something went wrong.");
+            }
+        }
+
+        internal void GetDrinks(string choice)
```
Comment (ananttuli) [inline Drinks.kjanos89/Drinks.kjanos89/Drinks.kjanos89/ApiController.cs:30]: Any particular reason this is marked as `internal` ?

File: Drinks.kjanos89/Drinks.kjanos89/Drinks.kjanos89/ApiController.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                List<Drink> drinks = serialize.DrinkList;
+                Menu.ShowData(drinks, "Drinks");
+
+            }
+            else
+            {
+                Console.WriteLine("Something went wrong.");
+            }
+        }
+        public void GetIngredients(string drink)
```
Comment (ananttuli) [inline Drinks.kjanos89/Drinks.kjanos89/Drinks.kjanos89/ApiController.cs:48]: Not a major issue, but I found some places where code conventions (https://thecsharpacademy.com/article/58/code-conventions) could be followed e.g. spaces between functions, namespace definition style, removing unnecessary whitespace etc.

**Review by ananttuli — CHANGES_REQUESTED** [review]
@kjanos89 Generally works 👍 but unexpected user input and exceptions could be handled more gracefully:

**Categories list screen**
- If the incorrect category name is entered, the program crashes with an error `Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.` instead of asking the user for the category input again.

**Drinks list screen (after selecting a category)**
- If a wrong numerical ID is entered e.g. `9999`, the program crashes with `Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.`

- If a wrong non-numeric drink ID is entered at first, e.g. `abc123`, the user remains stuck (even if they put in the correct ID or press '0' to go back)

```bash
----------------------------------
| Drinks |                       |
----------------------------------
| 15423  | 110 in the shade      |
----------------------------------

Choose a drink from the list below or press '0' to go back to the Categories menu:
abc123
Try again..
15423
Try again..
0
Try again..
stuck
Try again..

```

Since you were following the video tutorial, just ICYMI and only if you need guidance, here's the link to the full source code from the video:  https://github.com/cappuccinocodes/CSharpAcademy.Console.DrinksInfo

**Comment by kjanos89** [comment]
Hey, @ananttuli ! Thanks for reviewing my code. I apologize for missing all the input validation, it is fixed now.
Thank you for pointing out my lack of code quality, from now on I will pay more attention to follow the code-convention rules. Tried to follow them with this project as well now that I had to fix it.
As for the source code I generally dislike to watch others solve problems or telling me how to solve it without me giving it a go first, but for some reason Pablo talks in such a clear way I decided to watch the video and followed it along. No excuses, I should have handled all input cases on my own.
Once again, thank your for your time spent with reviewing my code.

**Review by ananttuli — APPROVED** [review]
@kjanos89 Approved ✅ Thanks for implementing the fixes. Inputs are handled gracefully now and adherence to code conventions is much improved.

>As for the source code I generally dislike to watch others solve problems or telling me how to solve it without me giving it a go first, but for some reason Pablo talks in such a clear way I decided to watch the video and followed it along.

Really glad to hear that. I encourage and adopt the same approach myself.

 All the best for the next one👍

**Pull request merged**